From d6716ad29fd2412989bcac1afc3e6cf67f0874bb Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 2 Oct 2024 13:04:44 +0100
Subject: [PATCH 635/699] drm/vc4: fkms: No SMI abuse needed on BCM2712

Since we don't use the (absent) SMI block to create interrupts on
BCM2712, there's no need to map any registers.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 drivers/gpu/drm/vc4/vc4_firmware_kms.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/drivers/gpu/drm/vc4/vc4_firmware_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
@@ -2010,16 +2010,16 @@ static int vc4_fkms_bind(struct device *
 	}
 
 	if (num_displays > 0) {
-		/* Map the SMI interrupt reg */
-		crtc_list[0]->regs = vc4_ioremap_regs(pdev, 0);
-		if (IS_ERR(crtc_list[0]->regs))
-			DRM_ERROR("Oh dear, failed to map registers\n");
-
 		if (fkms->revision >= BCM2712) {
 			ret = devm_request_irq(dev, platform_get_irq(pdev, 0),
 					       vc4_crtc2712_irq_handler, 0,
 					       "vc4 firmware kms", crtc_list);
 		} else {
+			/* Map the SMI interrupt reg */
+			crtc_list[0]->regs = vc4_ioremap_regs(pdev, 0);
+			if (IS_ERR(crtc_list[0]->regs))
+				DRM_ERROR("Oh dear, failed to map registers\n");
+
 			writel(0, crtc_list[0]->regs + SMICS);
 			ret = devm_request_irq(dev, platform_get_irq(pdev, 0),
 					       vc4_crtc_irq_handler, 0,
